src/braket/default_simulator/density_matrix_simulator.py

```diff
@@ -12,24 +12,38 @@
 # language governing permissions and limitations under the License.
 
 import sys
+from typing import Union
 
 from braket.device_schema.simulators import (
     GateModelSimulatorDeviceCapabilities,
     GateModelSimulatorDeviceParameters,
 )
+from braket.ir.jaqcd import Program as JaqcdProgram
+from braket.ir.openqasm import Program as OQ3Program
+from braket.task_result import GateModelTaskResult
 
 from braket.default_simulator.density_matrix_simulation import DensityMatrixSimulation
-from braket.default_simulator.jaqcd_simulator import BaseLocalJaqcdSimulator
+from braket.default_simulator.jaqcd_density_matrix_simulator import JaqcdDensityMatrixSimulator
+from braket.default_simulator.openqasm._helpers.utils import singledispatchmethod
+from braket.default_simulator.openqasm_density_matrix_simulator import (
+    OpenQASMDensityMatrixSimulator,
+)
 
 
-class DensityMatrixSimulator(BaseLocalJaqcdSimulator):
+class DensityMatrixSimulator(OpenQASMDensityMatrixSimulator, JaqcdDensityMatrixSimulator):
```

Comment:
A bit concerned about the multiple inheritance here (an in general, to be fair; probably makes my thoughts on OOP obvious). In particular, we're running into the [diamond problem](https://en.wikipedia.org/wiki/Multiple_inheritance#The_diamond_problem) here, since both parent classes subclass [BaseLocalSimulator](https://github.com/aws/amazon-braket-default-simulator-python/blob/local-sim-branch/src/braket/default_simulator/simulator.py#L55).

Can we do some kind of dispatch instead?

Reply:
Yes, that's how I initially implemented this, but I thought multiple inheritance was a bit neater. Maybe we can chat offline to see what makes more sense